wrappers/Python/CoolProp/AbstractState.pxd

```diff
@@ -156,10 +156,13 @@ cdef class AbstractState:
     cpdef double helmholtzmolar_excess(self) except *
     cpdef double helmholtzmass_excess(self) except *
 
-    
+    cpdef double gibbsmolar_residual(self) except *
+    cpdef double hmolar_residual(self) except *
+    cpdef double smolar_residual(self) except *
```

Comment:
I think they also need to get added to cAbstractState.pxd . This code duplication for C++ classes is one reason that I much prefer pybind11 for wrapping C++.  Perhaps you can test compilation for Python?

Reply:
I added them to cAbstractState.pxd as well. I tested compilation and ran a small Python script to check that it worked. The PC-SAFT values matched those from the HEOS backend.